torch/_inductor/decomposition.py

```diff
@@ -212,8 +212,19 @@ def mm(self, input2):
 
 @register_decomposition([aten.cat.default])
 def cat(tensors, dim=0):
-    if len(tensors) == 1:
+    def non_empty_tensor(x):
+        # special case for cat'ing with an empty tensor -
+        # just drop the 'empty' inputs so they don't confuse the logic below.
```

Comment:
I'm trying to understand in what sense you want to avoid confusing the logic below; and my best guess is that you're actually referring to torch/_inductor/fx_passes/split_cat.py (where you are enforcing an invariant that there are no malformed inputs to cat.) Is this correct?

Reply:
I think the comment is refering to the lowering for `aten.cat` which doesn't have the special case for empty 1d tensors. In eager tensors with shape `[0]` are ignored completely by the cat operator, and don't need to have the correct shape in the non-dim dimensions or even the correct number of dimensions.

Reply:
Hmm, but this seems questionable, we should be on the hook for accurately lowering aten.cat as it is defined. If we wanted a simplified version of it, we ought to have a prim... which we do! It's called prims.cat.